frb_codegen/src/generator/dart/mod.rs

```diff
@@ -521,3 +564,47 @@ fn dart_metadata(metadata: &[IrDartAnnotation]) -> String {
     }
     metadata
 }
+
+// Tests if a given struct has methods, that is, if the `ir_file` contains
+// a function that receives the struct as first argument
+fn struct_has_methods(file: &IrFile, the_struct: Option<&IrType>) -> bool {
+    let the_struct = if let Some(s) = the_struct {
+        s
+    } else {
+        return false;
+    };
+    if the_struct.safe_ident().contains("user_id") {
+        println!(
+            "struct_has_methods for file: {:?}, the_struct: {:?}",
+            file, the_struct
+        );
+    }
+    let struct_name = if let StructRef(IrTypeStructRef { name, freezed: _ }) = the_struct {
+        name
+    } else {
+        return false;
+    };
+
+    file.funcs.iter().any(|f| {
+        if let Some(IrField {
+            ty: _,
+            name,
+            is_final: _,
+            comments: _,
+        }) = f.inputs.get(0)
+        {
+            if name.raw.to_case(Case::UpperCamel) == *struct_name {
```

Comment:
Well, do we really need to do such comparsion, especially with a re-casing? That sounds brittle...

Reply:
We need to find the function for which the first argument is the struct. I don't see other way

Reply:
All right. Then maybe we can add a comment about this